### PR TITLE
Don't break on paired shortcodes

### LINKED ISSUE #5902: Trouble escaping a Hugo paired shortcode
### Bug description

We have a Quarto book where we document how to write blog posts on our Hugo website. With the pre-release version of Quarto (and the released version, but not with `1.3.49` that I have locally) the build fails because of the chunk below:

````
```{shortcodes=false}

{{< gallery >}}
{{< figureforgallery src="search2.png" alt="Screenshot of r-universe simple search results." >}}
{{< figureforgallery src="search3.png" alt="Screenshot of r-universe advanced search query." >}}
{{< figureforgallery src="search4.png" alt="Screenshot of r-universe advanced search results." >}}
{{< /gallery >}}
```
````

The error is `ERROR: invalid shortcode: /gallery`.

### Steps to reproduce

I've set up a reprex at https://github.com/maelle/quartohugo It's the default Quarto book created with RStudio IDE (Create New Project > Quarto book) except I tweaked https://github.com/maelle/quartohugo/blob/main/index.qmd to include the faulty chunk.

I added a GHA workflow https://github.com/maelle/quartohugo/blob/main/.github/workflows/render.yaml The latest log is at https://github.com/maelle/quartohugo/actions/runs/5253508823/jobs/9490928013

### Expected behavior

The book should be built and the shortcode should appear as is in code.

### Actual behavior

There is an error `ERROR: invalid shortcode: /gallery`.

### Your environment

ubuntu-latest on GHA

### Quarto check output

I'm using the official Quarto GitHub Action so hopefully it's fine I don't answer this!

### PR DESCRIPTION
**Pull request by cscheid**
This closes #5902.

### CODE REVIEW

**Comment by cscheid** [comment]
(This bug also exists in 1.3, I think it's a clean candidate for backporting)

File: src/core/lib/parse-shortcode.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export function isBlockShortcode(content: string) {
 
 function parseShortcodeCapture(capture: string): Shortcode | undefined {
   // match shortcode name
-  const nameMatch = capture.match(/^[a-zA-Z0-9_]+/);
+  const nameMatch = capture.match(/^\/?[a-zA-Z0-9_]+/);
```
Comment (cderv) [inline src/core/lib/parse-shortcode.ts:25]: This does means we now support a name starting with `/` (e.g `{{< /name >}}`), right ? 

This part of the code is not just about escaping shortcode so I am wondering.

**Review by cderv — COMMENTED** [review]
Thanks Carlos ! 

As the initial issue is about escaping shortcodes , and not adding support for `/` in shortcode, I am wondering if changing this is more broad. 

I understood from your previous comment that we can't really detect an escaped shortcode inside a  code block with specific attributes. Did I got it right ? 

If so it seems best effort to not trip on such syntax. However, this could lead us to support more specific case in the future. 

One of the comment in the PR is mentionning, teaching about unsupported shortcode for example, and this fix only won't solve it. 

````markdown
---
format: html
---

```{shortcodes=false}
# Shortcode syntax
{{< test something >}}
# don't start with weird characters
{{< ~test something >}}
```

<!-- Using {{{ }}} works for escaping -->
```
# don't start with weird characters
{{{< ~test something >}}}
```
````

I am just wondering if this should be broader somehow, also seeing your code comment
https://github.com/quarto-dev/quarto-cli/blob/9242a9c475041b2307b1b84044bc79ca81e1cd69/src/core/lib/parse-shortcode.ts#L87-L88

File: src/core/lib/parse-shortcode.ts (hunk 1 above)
Comment (cscheid) [reply src/core/lib/parse-shortcode.ts:25]: > This does means we now support a name starting with / (e.g {{< /name >}}), right ?

That's right. As far as I can tell, other non-alphanumeric characters besides `/` aren't valid first characters fo Hugo shortcodes (although I didn't actually find a spec for them)

Comment (cderv) [reply src/core/lib/parse-shortcode.ts:25]: Yes I think `/` is the only one at least obvious, for closing shortcode. 

Probably allowing this for our shortcode name won't be that bad, especially if not documented, no one will build a shortcode with leading `/` 😅 

> One of the comment in the PR is mentionning, teaching about unsupported shortcode for example, and this fix only won't solve it.

Still about the issue, this PR does not solve this then. But I believe we can document the limitation as `{{{< ... >}}}` escaping works; So it seems ook.

Comment (cscheid) [reply src/core/lib/parse-shortcode.ts:25]: Wait. I'm sorry if I'm being stupid again, but does the smoke test I added not work for you?

Comment (cderv) [reply src/core/lib/parse-shortcode.ts:25]: No no all good on your side. I am just pushing on some edge case. As shared above, I was thinking of some tutorial document where you would show an unsupported shortcode syntax like below. 

````
```{shortcodes=false}
# Shortcode syntax
{{< test something >}}
# don't start with weird characters
{{< ~test something >}}
```
````

As a user, reading the doc, you would expect `shortcodes=false` to be escaping the shortcodes, so not 'processing' them. But this would throw an error. 

Anyhow, I understand now better how we handle shortcodes. 

Conclusion is: 

* `{{{< ... >}}}` works for escaping any shortcode name (valid or not valid)
* `` ```{shortcodes=false} ``  will only works on valid shortcode name, because we are parsing and check shortcode validity anyway. 

So as I said, this is a edge case. 

This PR can be merge to support closing shortcode syntax from hugo.
